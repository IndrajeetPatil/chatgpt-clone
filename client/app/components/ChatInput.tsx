import { useState } from "react";
import TextField from "@mui/material/TextField";
import Button from "@mui/material/Button";
import SendIcon from "@mui/icons-material/Send";
import Box from "@mui/material/Box";

interface ChatInputProps {
  onSendMessage: (message: string) => void;
}

export default function ChatInput({ onSendMessage }: ChatInputProps) {
  const [message, setMessage] = useState("");

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (message.trim()) {
      onSendMessage(message);
      setMessage("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <Box sx={{ display: "flex", alignItems: "flex-start" }}>
        <TextField
          fullWidth
          variant="outlined"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
          placeholder="Type your message..."
          multiline
          rows={2}
          sx={{ mr: 1 }}
        />
        <Button
          type="submit"
          variant="contained"
          endIcon={<SendIcon />}
          sx={{ height: "100%" }}
        >
          Send
        </Button>
      </Box>
    </form>
  );
}
