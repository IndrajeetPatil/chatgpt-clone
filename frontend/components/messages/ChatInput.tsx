import React, { KeyboardEvent, useState } from "react";

import SendIcon from "@mui/icons-material/Send";
import { Box, IconButton, TextField } from "@mui/material";

interface ChatInputProps {
  onSendMessage: (message: string) => void;
}

const ChatInput: React.FC<ChatInputProps> = ({ onSendMessage }) => {
  const [message, setMessage] = useState("");

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    sendMessage();
  };

  const sendMessage = () => {
    if (message.trim()) {
      onSendMessage(message.trim());
      setMessage("");
    }
  };

  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
    if (event.key === "Enter" && !event.shiftKey) {
      event.preventDefault();
      sendMessage();
    }
  };

  return (
    <Box
      component="form"
      onSubmit={handleSubmit}
      display="flex"
      alignItems="flex-end"
      mt={2}
    >
      <TextField
        multiline
        fullWidth
        placeholder="Type your message..."
        rows={2}
        value={message}
        onChange={(e) => setMessage(e.target.value)}
        onKeyDown={handleKeyDown}
        sx={{ mr: 1 }}
      />
      <IconButton
        type="submit"
        color="primary"
        disabled={!message.trim()}
        aria-label="Send message"
      >
        <SendIcon />
      </IconButton>
    </Box>
  );
};

export default ChatInput;
