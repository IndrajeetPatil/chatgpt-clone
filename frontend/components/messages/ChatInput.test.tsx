import "@testing-library/jest-dom";

import { fireEvent, render } from "@testing-library/react";

import ChatInput from "./ChatInput";

describe("ChatInput component", () => {
  it("should render the ChatInput component and match the snapshot", () => {
    const onSendMessageMock = jest.fn();
    const { asFragment } = render(
      <ChatInput onSendMessage={onSendMessageMock} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should call onSendMessage with the message when Enter is pressed", () => {
    const onSendMessageMock = jest.fn();
    const { getByPlaceholderText } = render(
      <ChatInput onSendMessage={onSendMessageMock} />
    );

    const input = getByPlaceholderText("Type your message...");
    fireEvent.change(input, { target: { value: "Hello, World!" } });
    fireEvent.keyDown(input, { key: "Enter", code: "Enter", charCode: 13 });

    expect(onSendMessageMock).toHaveBeenCalledWith("Hello, World!");
  });

  test("should not call onSendMessage if Enter is pressed with an empty message", () => {
    const onSendMessageMock = jest.fn();
    const { getByPlaceholderText } = render(
      <ChatInput onSendMessage={onSendMessageMock} />
    );

    const input = getByPlaceholderText("Type your message...");
    fireEvent.keyDown(input, { key: "Enter", code: "Enter", charCode: 13 });

    expect(onSendMessageMock).not.toHaveBeenCalled();
  });
});
