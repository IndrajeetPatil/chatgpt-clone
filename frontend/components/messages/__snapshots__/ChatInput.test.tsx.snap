// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput component should render the ChatInput component and match the snapshot 1`] = `
<DocumentFragment>
  <form
    class="MuiBox-root css-mkkhhh"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-15e2bhn-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-zsp0bh-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          placeholder="Type your message..."
          rows="2"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-w4cd9x"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      aria-label="Send message"
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-186dzrf-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="SendIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
